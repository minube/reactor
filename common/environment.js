import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
