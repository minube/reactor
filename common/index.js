import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
