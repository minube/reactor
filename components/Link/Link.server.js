import { createElement } from 'react';

export default ({ href, ...inherit }) => createElement('a', { href }, inherit.children);
