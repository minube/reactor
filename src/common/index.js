import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
