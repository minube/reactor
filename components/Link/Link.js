import { string } from 'prop-types';
import React, { PureComponent } from 'react';

import { THEME } from '../../common';
import Text from '../Text';

class Link extends PureComponent {
  state = {
    hover: false,
  }

  render() {
    const {
      props: {
        href, ...inherit
      },
      state: { hover },
    } = this;

    return (
      <Text
        {...inherit}
        accessibilityRole="link"
        href={href}
        color={hover && !inherit.styleHover ? THEME.COLOR.ACCENT : undefined}
        onMouseEnter={() => this.setState({ hover: true })}
        onMouseLeave={() => this.setState({ hover: false })}
        style={[inherit.style, hover && inherit.styleHover]}
      />
    );
  }
}

Link.propTypes = {
  href: string.isRequired,
};

export default Link;
