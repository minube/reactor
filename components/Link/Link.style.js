import { StyleSheet } from 'react-native';

import { THEME } from '../../common';

const {
  COLOR,
} = THEME;

export default StyleSheet.create({
  hover: {
    color: COLOR.ACCENT,
  },
});
