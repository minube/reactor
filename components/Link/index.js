import { ENV } from '../../common';
import Default from './Link';

const Loading = ENV.IS_SERVER ? require('./Link.server').default : Default;

export default Loading;
