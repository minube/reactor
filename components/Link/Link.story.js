import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import Link from './Link';

storiesOf('✅ Link', module)
  .addWithJSX('default', () => (
    <Link href="http://minube.com">minube.com</Link>
  ))
  .addWithJSX('without href', () => (
    <Link>minube.com</Link>
  ))
  .addWithJSX('style', () => (
    <Link
      href="http://minube.com"
      style={{ fontSize: THEME.FONT.SIZE.LARGE, fontWeight: THEME.FONT.WEIGHT.BOLD }}
      styleHover={{ color: 'red' }}
    >
      minube.com
    </Link>
  ))
  .addWithJSX('🏀 Playground', () => (
    <Link href={text('href', 'http://minube.com')}>
      {text('title', 'minube.com')}
    </Link>
  ));
