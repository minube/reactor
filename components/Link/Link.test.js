import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import Link from './Link';

describe('<Link>', () => {
  it('renders', () => {
    const tree = renderer.create(<Link />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {href}', () => {
    const tree = renderer.create(<Link href="http://minube.com" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const style = { fontSize: THEME.FONT.SIZE.LARGE, fontWeight: THEME.FONT.WEIGHT.BOLD };
    const tree = renderer.create(<Link visible style={style} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {styleHover}', () => {
    const styleHover = { fontSize: THEME.FONT.SIZE.LARGE, fontWeight: THEME.FONT.WEIGHT.BOLD };
    const tree = renderer.create(<Link visible styleHover={styleHover} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Link />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

