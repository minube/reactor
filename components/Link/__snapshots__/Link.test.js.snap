// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> renders 1`] = `
<Text
  accessibilityRole="link"
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  href={undefined}
  numberOfLines={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
      },
      false,
      false,
      false,
      false,
      Object {
        "fontSize": 16,
        "lineHeight": 20.8,
      },
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      Object {},
    ]
  }
/>
`;

exports[`<Link> when {href} 1`] = `
<Text
  accessibilityRole="link"
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  href="http://minube.com"
  numberOfLines={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
      },
      false,
      false,
      false,
      false,
      Object {
        "fontSize": 16,
        "lineHeight": 20.8,
      },
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      Object {},
    ]
  }
/>
`;

exports[`<Link> when {style} 1`] = `
<Text
  accessibilityRole="link"
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  href={undefined}
  numberOfLines={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
      },
      false,
      false,
      false,
      false,
      Object {
        "fontSize": 16,
        "lineHeight": 20.8,
      },
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      },
    ]
  }
  visible={true}
/>
`;

exports[`<Link> when {styleHover} 1`] = `
<Text
  accessibilityRole="link"
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  href={undefined}
  numberOfLines={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Array [
      Object {
        "color": "#4a4a4a",
        "fontFamily": undefined,
      },
      false,
      false,
      false,
      false,
      Object {
        "fontSize": 16,
        "lineHeight": 20.8,
      },
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      Object {},
    ]
  }
  styleHover={
    Object {
      "fontSize": 20,
      "fontWeight": "bold",
    }
  }
  visible={true}
/>
`;
